src/components/common/PeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -730,20 +730,11 @@ export default class PeopleAndRoles extends Mixins(CommonMixin, DateMixin, OrgPe
         if (deleted) delete deleted.actions
       }
     } else {
-      // get ID of edited person to undo
-      const id = person?.officer?.id
-
-      let thisPerson
-      if (isNaN(+id)) {
-        // to check assigned UUID
-        thisPerson = cloneDeep(this.originalParties.find(x => x.officer.id === id))
-      } else {
-        // get a copy of original person from original IA
-        thisPerson = cloneDeep(this.originalParties.find(x => +x.officer.id === +id))
-      }
+      // copy the original person
+      let thisPerson = cloneDeep(this.originalParties[0])
 
```

Comment:
The previous codes won't work because this.originalParties[0]?.officer?.id (the id) actually would change, every time when we save and resume. So we can copy the original person directly and splice it back.

Reply:
When we save and resume a draft, the JSON should be untouched by the Legal API. Do you have an example of the saved JSON being different from the resumed JSON?

Reply:
I'm not sure values the old code above is comparing, but if I look in a sample filing, I can see the Removed party with its ID that should be unchanged.

![image](https://github.com/user-attachments/assets/f555f40d-3ca1-4205-810f-66aecd761222)

Reply:
I tried with this company:
https://dev.business.bcregistry.gov.bc.ca/BC0883552/?accountid=3040
This is the officer.id when we start a new extension application, with it never being save and resumed:
![image](https://github.com/user-attachments/assets/b52c0880-3127-47e2-a816-25a1d1c5671b)
After we save and resume it, it changes:
![image](https://github.com/user-attachments/assets/a48e9d52-6ec3-47d0-9537-9e308d067c14)
I used console log to see the same changes

Reply:
the id only changes when the application is never save and resumed, and after it gets saved and resumed; the later save and resumes won't change it again

Reply:
This is where I try console.log: 
![image](https://github.com/user-attachments/assets/e4b539ba-0a0f-4103-b16a-8d693ab4cacf)

Open a new application, never save and resumed:
![image](https://github.com/user-attachments/assets/78bffde7-d40d-4efe-825a-34ad953966ce)

After save and resumed:
![image](https://github.com/user-attachments/assets/fbead802-fff5-4675-9cf2-16eecf3821cf)

Reply:
Resuming should give you the exact same data you saved. The Legal API does not look at or change anything.

Possibly, above, the id is refreshed by the UI on save? Or the wrong object is being saved?

I can look at it on Monday if you want.

Reply:
When I save your existing Limited Restoration Extension, the data is:

![image](https://github.com/user-attachments/assets/6cb1698e-535e-4527-9c88-d8f23b7977e1)

On resume, I see the data as:

![image](https://github.com/user-attachments/assets/053a91c5-bf61-4d4c-857b-4a6ba1767563)

The party officer IDs are the same. If you're seeing something different then something else is going on.

(PS Make sure you look at the payload / preview for the LRE filing (150953), not the "state filing" (150917) that is also fetched.)

Reply:
PS I deleted and re-created the Limited Restoration Extension filing, so it's a new filing id now.

Reply:
Thanks Sev. I figured out this is because the LimitedRestorationExtension Vue and LimitedRestorationToFull Vue did a re-assign of officer ID. Please see below.
I removed that re-assign so this bug is fixed.

Reply:
Let's close this conversation and I will open a new conversation in your new code...